apiserver/facades/agent/provisioner/provisioninginfo.go

```diff
@@ -63,15 +63,12 @@ func (api *ProvisionerAPI) ProvisioningInfo(ctx context.Context, args params.Ent
 
 	for i, entity := range args.Entities {
 		tag, err := names.ParseMachineTag(entity.Tag)
-		if err != nil {
+		if err != nil || !canAccess(tag) {
```

Comment:
We still need a `canAccess` check right?

Reply:
Yes!